File: examples/src/main/java/webshop/ReserveItemsInstance.java
package webshop;

import java.util.List;

public class ReserveItemsInstance implements JWebshopRecipe.ReserveItems {

    @Override
    public ReserveItemsOutcome apply(String id, List<String> items) {
        return new ItemsReserved(items);
    }
}

File: examples/baker-example/src/main/scala/webshop/webservice/ReserveItemsInstance.scala
package webshop.webservice

import cats.effect.IO
import webshop.webservice.CheckoutFlowEvents.ReserveItemsOutput
import webshop.webservice.CheckoutFlowIngredients.{Item, ReservedItems}
import webshop.webservice.CheckoutFlowInteractions.ReserveItems

import scala.concurrent.Future
import scala.concurrent.duration._
import cats.effect.Temporal

class ReserveItemsInstance(implicit timer: Temporal[IO]) extends ReserveItems {

  override def apply(items: List[Item]): Future[ReserveItemsOutput] = {
      IO.sleep(1.second)
        .as(CheckoutFlowEvents.ItemsReserved(ReservedItems(items, Array.fill(1000)(Byte.MaxValue))))
        .unsafeToFuture()
  }
}
